src/Paramore.Brighter/InMemoryInbox.cs

```diff
@@ -104,22 +180,26 @@ public class InMemoryInbox : IAmAnInbox, IAmAnInboxAsync
         /// <exception cref="System.TypeLoadException"></exception>
         public T Get<T>(Guid id, string contextKey, int timeoutInMilliseconds = -1) where T : class, IRequest
         {
+            ClearExpiredMessages();
+            
             if (!Exists<T>(id, contextKey))
             {
                throw new RequestNotFoundException<T>(id);
             }
 
-            var inboxItem = _commands[CreateKey(id, contextKey)];
-            if (inboxItem.CommandType != typeof (T))
-                throw new TypeLoadException(string.Format($"The type of item {id} is {inboxItem.CommandType.Name} not {typeof(T).Name}"));
+            var inboxItem = _requests[InboxItem.CreateKey(id, contextKey)];
```

Comment:
Probably use _requests.TryGet() so you can handle found/not found in one operation?

Reply:
I'll peek at it. Thanks

Reply:
OK, implemented that, thanks for the suggestion